cpp/memilio/io/result_io.cpp

```diff
@@ -89,6 +89,118 @@ IOResult<void> save_result(const std::vector<TimeSeries<double>>& results, const
     }
     return success();
 }
+IOResult<void> save_edges(const std::vector<std::vector<TimeSeries<double>>>& ensemble_edges,
+                          const std::vector<std::pair<int, int>>& pairs_edges, const fs::path& result_dir,
+                          bool save_single_runs, bool save_percentiles)
```

Comment:
Why do we need these functions? Since they also save a (collection of) TimeSeries, couldn't we use the save_result functions with a different file name?

Reply:
I tried that first. However, there were some problems. The ids are given by pairs and not by pure ints. In addition, the groups are now the start edges. The next level is then a group with the name "End", which denotes the end edge. These are not necessarily sequential as with age groups. (E.g. Could be 1002, 1009,.. and not 1,2,3,4...)
In addition, there were several small things that should have been adjusted.

 As the function is already very complex and difficult to understand, I decided to go without further abstractions and write a separate function for the edges.